core/shared/src/main/scala/scalaz/zio/Queue.scala

```diff
@@ -347,7 +347,7 @@ object Queue {
                   unsafeMovePutters()
                 } else {
                   unsafeOfferAll(putters, putter :: unsafePollAll(putters))
-                  ()
+                  if (!queue.isFull()) unsafeMovePutters() else ()
```

Comment:
I ran this fix on the code example I shared in the original issue and it looks like I got the issue again after about 4.5 million runs, 5 million runs and 2.5 million runs. Although I can't say with absolute certainty that its the same race condition we're dealing with here.

Not that this would eliminate the race condition, but `unsafeMovePutters` already has an `isFull` check at the beginning, could probably get remove this one?

Reply:
UPDATE: I am seeing another race condition with different circumstances than this one. Will create an issue after I have more information.